packages/commerce-sdk-react/src/hooks/useUsid.ts

```diff
@@ -5,27 +5,30 @@
  * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
  */
 import useAuthContext from './useAuthContext'
-import useLocalStorage from './useLocalStorage'
-import useConfig from './useConfig'
 
-const onClient = typeof window !== 'undefined'
+/**
+ * @group Shopper Authentication helpers
+ */
+interface Usid {
+    usid: string | null
+    getUsidWhenReady: () => Promise<string>
+}
 
 /**
  * Hook that returns the usid associated with the current access token.
  *
  * @group Helpers
  * @category Shopper Authentication
  */
-const useUsid = (): string | null => {
-    const config = useConfig()
+const useUsid = (): Usid => {
     const auth = useAuthContext()
+    const usid = auth.get('usid')
+
+    // NOTE: auth.ready() is to be called later. If you call it immediately in this hook,
+    // it'll cause infinite re-renders during testing.
+    const getUsidWhenReady = () => auth.ready().then(({usid}) => usid)
 
-    return onClient
-        ? // This conditional is a constant value based on the environment, so the same path will
-          // always be followed., and the "rule of hooks" is not violated.
-          // eslint-disable-next-line react-hooks/rules-of-hooks
-          useLocalStorage(`usid_${config.siteId}`)
-        : auth.get('usid')
```

Comment:
why is the `onClient` ternary remove? is this intentional?

Reply:
`auth.get('usid')` doesn't trigger re-renders, so app will be broken on client side

Reply:
I removed the ternary because for usid, `useLocalStorage` will always return null since the `usid` is in a cookie and not local store.

Reply:
this works for now in retail react app because when you log out (and get a new `usid`), you always get redirected to login page... and cause the components to unmount.

if there is a way to log out on PDP and stays on that page, then we will have a trouble.

We can fix this later, seems like a non blocking issue.

Reply:
But when would the usid cookie change? The only cases I can think of is when someone deletes or manipulates the usid cookie directly.

Otherwise the usid is static as once a shopper lands on the site, SLAS will generate a usid and that shopper will continue to use the same usid through login (if we don't cart merge will not work).

Reply:
On 2nd thought, logout could change the usid since that creates a new guest user 🤔

Reply:
I've added a TODO on useUsid regarding the logout case.